openedx/core/djangoapps/courseware_api/views.py

```diff
@@ -72,6 +72,7 @@ def __init__(self, course_key, request, username=''):
             course_key,
             staff_access=self.original_user_is_staff,
         )
+        self.request.user = self.effective_user
```

Comment:
@mikix i guess that should work, since we're still recording `original_user_is_staff`; are there any details on what exactly wasn't working? just curious since i thought it had been _generally_ working, so curious what/where the edge cases were...

Reply:
Waffles. They get the current request and use that to check if the current user should have the waffle on or not: